test-utils/src/lib.rs

```diff
@@ -47,23 +49,32 @@ pub async fn init_core_and_create_wf(test_name: &str) -> (Arc<dyn Core>, String)
     (core, starter.get_task_queue().to_string())
 }
 
-/// Create a core instance that will use the provided history to return a workflow task containing
-/// all of it, for testing replay.
-pub async fn init_core_replay(history: &History) -> Arc<dyn Core> {
-    let mut starter = CoreWfStarter::new_tq_name(TEST_Q);
-    let core_fakehist = init_mock_gateway(
-        starter.core_options.clone(),
-        mock_gateway_from_history(history),
-    )
-    .unwrap();
-    core_fakehist
-        .register_worker(starter.worker_config.clone())
-        .await
-        .unwrap();
-    starter.initted_core = Some(Arc::new(core_fakehist));
-    let core = starter.get_core().await;
-    starter.start_wf().await;
-    core
+/// Create a core instance that can be used for replay. See the [ReplayCore] trait for adding
+/// canned history.
+pub fn init_core_replay(opts: TelemetryOptions) -> ReplayCoreImpl {
+    let shared_mock_gateway = mock_gateway();
+    let init_opts = CoreInitOptionsBuilder::default()
+        .gateway_opts(shared_mock_gateway.get_options().clone())
+        .telemetry_opts(opts)
```

Comment:
Should the replay core even emit telemetry?

Reply:
Yes, why not? It's configurable in the same way and it's necessary for console output. It doesn't automatically mean exporting is on, and langs can default it to off when constructing

Reply:
Sure for logging that makes sense.
We could limit the supported options on the rust side to be extra safe but I'm fine with limiting this on the lang side too.
For the TS PR we'd need to provide a way to expose `ReplayCore` in order to configure logging which I was hoping we could avoid.